src/app/blog/faceting/index.mdx

```diff
@@ -0,0 +1,287 @@
+
+In this post, we'll break down how we brought **Elasticsearch-style faceting directly into PostgreSQL;** the syntax, the planner integration, and the execution strategy that makes it possible. We'll show you how we structured the SQL API using window functions, and the results we can get from pushing as much work as possible down into our underlying search library (spoiler alert: [over an order of magnitude faster](#performance-why-this-matters) for large result sets).
+
+<Note>
+  If you aren't familiar with [ParadeDB](https://paradedb.com), we're a Postgres
+  extension that brings modern search capabilities directly into your database.
+  Instead of managing separate search infrastructure, you get BM25 full-text
+  search, vector search, and real-time analytics while keeping all your data in
+  Postgres with full ACID guarantees.
+</Note>
+
+## How Search Engines Count
+
+Faceting is a way of summarizing the contents of a search result. When you type a query like "dinosaur", you’re not just asking for a list of matching documents, you’re also thinking, _what kinds of dinosaurs exist in this result set?_ Maybe there are 87 carnivores, 112 herbivores, and 41 omnivores. Each item in that breakdown, the counts of results grouped by attribute, is a **facet**.
+
+Facets turn search from a one-dimensional list into something that can help visualize navigation. They set the stage to let you explore results by filtering on structured fields (like _era_, _habitat_, or _height range_) while still searching free text (like _large dinosaur_). In other words, faceting is where structured data meets unstructured search.
+
+<figure className="flex flex-col items-center">
+  <Image
+    src={faceting}
+    alt="Faceted search powering real-world insights: an example from our customer, DemandScience."
+    className="rounded-xl"
+  />
+  <figcaption className="mt-2 text-sm text-gray-500">
+    Faceted search powering real-world insights: an example from our customer,
+    [DemandScience](https://demandscience.com/).
+  </figcaption>
+</figure>
+
+Behind the scenes, each facet corresponds to a column or field in your dataset. A search engine computes how many matching documents belong to each unique value or bucket of that field, then returns both the top results and these aggregate counts in a single response. Because a search result usually returns results that are spread across our dataset the database needs a quick way to look up values for each document.
+
+That duality, returning ranked search hits _and_ category counts, is what makes faceting so powerful. It's also what makes it tricky to implement efficiently in a database built for rows, not documents.
```

Comment:
I don't think that document databases make this any easier... analytical databases do though? This might be more about the traditional OLTP/OLAP dichotomy.